include/aws/io/stream.h

```diff
@@ -38,17 +39,47 @@ struct aws_input_stream_vtable {
     int (*read)(struct aws_input_stream *stream, struct aws_byte_buf *dest);
     int (*get_status)(struct aws_input_stream *stream, struct aws_stream_status *status);
     int (*get_length)(struct aws_input_stream *stream, int64_t *out_length);
-    void (*destroy)(struct aws_input_stream *stream);
+
+    /**
+     * Optional. If not set, the default aws_ref_count_acquire/release will be used.
+     * Set for high level language binding that has its own refcounting implementation and needs to be kept alive from
+     * C.
+     * When the C level needs to keep the high level language binding alive, while binding as the user also holds the
+     * reference to the C object, it will result in a recursive reference.
+     * The function here is for C Object to use the high level refcount implementation directly.
```

Comment:
This sentence doesn't make a ton of sense, just remove it. previous sentence says it nicely

Reply:
Add comment about how the refcount isn't used if you override acquire()/release()